// @vertx/core/enums.mjs
/*
 * Copyright 2020 ES4X
 *
 * ES4X licenses this file to you under the Apache License, version 2.0
 * (the "License"); you may not use this file except in compliance with the
 * License.  You may obtain a copy of the License at:
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

/// <reference types="@vertx/core/enums" />

/**
 * @typedef { import("@vertx/core") } Java
 */
export const ClientAuth = Java.type('io.vertx.core.http.ClientAuth');
export const CookieSameSite = Java.type('io.vertx.core.http.CookieSameSite');
export const DnsResponseCode = Java.type('io.vertx.core.dns.DnsResponseCode');
export const HttpVersion = Java.type('io.vertx.core.http.HttpVersion');
export const JsonEventType = Java.type('io.vertx.core.parsetools.JsonEventType');
export const ProxyType = Java.type('io.vertx.core.net.ProxyType');
export const ReplyFailure = Java.type('io.vertx.core.eventbus.ReplyFailure');
export const TracingPolicy = Java.type('io.vertx.core.tracing.TracingPolicy');
export const WebsocketVersion = Java.type('io.vertx.core.http.WebsocketVersion');
