// @vertx/core/index.mjs
/*
 * Copyright 2020 ES4X
 *
 * ES4X licenses this file to you under the Apache License, version 2.0
 * (the "License"); you may not use this file except in compliance with the
 * License.  You may obtain a copy of the License at:
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

/// <reference types="@vertx/core" />

/**
 * @typedef { import("@vertx/core") } Java
 */
export const AsyncResult = Java.type('io.vertx.core.AsyncResult');
export const JsonObject = Java.type('io.vertx.core.json.JsonObject');
export const JsonArray = Java.type('io.vertx.core.json.JsonArray');

export const AsyncFile = Java.type('io.vertx.core.file.AsyncFile');
export const AsyncMap = Java.type('io.vertx.core.shareddata.AsyncMap');
export const Buffer = Java.type('io.vertx.core.buffer.Buffer');
export const CLI = Java.type('io.vertx.core.cli.CLI');
export const CommandLine = Java.type('io.vertx.core.cli.CommandLine');
export const Context = Java.type('io.vertx.core.Context');
export const Cookie = Java.type('io.vertx.core.http.Cookie');
export const Counter = Java.type('io.vertx.core.shareddata.Counter');
export const DatagramPacket = Java.type('io.vertx.core.datagram.DatagramPacket');
export const DatagramSocket = Java.type('io.vertx.core.datagram.DatagramSocket');
export const DeliveryContext = Java.type('io.vertx.core.eventbus.DeliveryContext');
export const DnsClient = Java.type('io.vertx.core.dns.DnsClient');
export const EventBus = Java.type('io.vertx.core.eventbus.EventBus');
export const FileProps = Java.type('io.vertx.core.file.FileProps');
export const FileSystem = Java.type('io.vertx.core.file.FileSystem');
export const FileSystemProps = Java.type('io.vertx.core.file.FileSystemProps');
export const HttpClient = Java.type('io.vertx.core.http.HttpClient');
export const HttpClientRequest = Java.type('io.vertx.core.http.HttpClientRequest');
export const HttpClientResponse = Java.type('io.vertx.core.http.HttpClientResponse');
export const HttpConnection = Java.type('io.vertx.core.http.HttpConnection');
export const HttpFrame = Java.type('io.vertx.core.http.HttpFrame');
export const HttpHeaders = Java.type('io.vertx.core.http.HttpHeaders');
export const HttpServer = Java.type('io.vertx.core.http.HttpServer');
export const HttpServerFileUpload = Java.type('io.vertx.core.http.HttpServerFileUpload');
export const HttpServerRequest = Java.type('io.vertx.core.http.HttpServerRequest');
export const HttpServerResponse = Java.type('io.vertx.core.http.HttpServerResponse');
export const JsonEvent = Java.type('io.vertx.core.parsetools.JsonEvent');
export const JsonParser = Java.type('io.vertx.core.parsetools.JsonParser');
export const JsonPointer = Java.type('io.vertx.core.json.pointer.JsonPointer');
export const JsonPointerIterator = Java.type('io.vertx.core.json.pointer.JsonPointerIterator');
export const LocalMap = Java.type('io.vertx.core.shareddata.LocalMap');
export const Lock = Java.type('io.vertx.core.shareddata.Lock');
export const Measured = Java.type('io.vertx.core.metrics.Measured');
export const Message = Java.type('io.vertx.core.eventbus.Message');
export const MessageConsumer = Java.type('io.vertx.core.eventbus.MessageConsumer');
export const MessageProducer = Java.type('io.vertx.core.eventbus.MessageProducer');
export const MultiMap = Java.type('io.vertx.core.MultiMap');
export const MxRecord = Java.type('io.vertx.core.dns.MxRecord');
export const NetClient = Java.type('io.vertx.core.net.NetClient');
export const NetServer = Java.type('io.vertx.core.net.NetServer');
export const NetSocket = Java.type('io.vertx.core.net.NetSocket');
export const Pipe = Java.type('io.vertx.core.streams.Pipe');
export const Promise = Java.type('io.vertx.core.Promise');
export const Pump = Java.type('io.vertx.core.streams.Pump');
export const ReadStream = Java.type('io.vertx.core.streams.ReadStream');
export const RecordParser = Java.type('io.vertx.core.parsetools.RecordParser');
export const SelfSignedCertificate = Java.type('io.vertx.core.net.SelfSignedCertificate');
export const ServerWebSocket = Java.type('io.vertx.core.http.ServerWebSocket');
export const SharedData = Java.type('io.vertx.core.shareddata.SharedData');
export const SocketAddress = Java.type('io.vertx.core.net.SocketAddress');
export const SrvRecord = Java.type('io.vertx.core.dns.SrvRecord');
export const StreamBase = Java.type('io.vertx.core.streams.StreamBase');
export const TimeoutStream = Java.type('io.vertx.core.TimeoutStream');
export const Vertx = Java.type('io.vertx.core.Vertx');
export const WebSocket = Java.type('io.vertx.core.http.WebSocket');
export const WebSocketBase = Java.type('io.vertx.core.http.WebSocketBase');
export const WebSocketFrame = Java.type('io.vertx.core.http.WebSocketFrame');
export const WorkerExecutor = Java.type('io.vertx.core.WorkerExecutor');
export const WriteStream = Java.type('io.vertx.core.streams.WriteStream');
